dcpy/connectors/github.py

```diff
@@ -1,25 +1,39 @@
+from __future__ import annotations
 import os
 import requests
-from dateutil.parser import parse
+from datetime import datetime
+from dateutil.parser import parse as datetime_parse
 from typing import List, Dict, Any
+from dataclasses import dataclass
 
 ORG = "NYCPlanning"
 PERSONAL_TOKEN = os.environ["GHP_TOKEN"]
 headers = {"Authorization": "Bearer %s" % PERSONAL_TOKEN}
 BASE_URL = f"https://api.github.com/repos/{ORG}"
 
 
-def parse_workflow(workflow: Dict) -> Dict:  ## json input
-    return {
-        "status": workflow["status"],
-        "conclusion": workflow["conclusion"],
-        "timestamp": parse(workflow["updated_at"], fuzzy=True),
-        "url": workflow["html_url"],
-    }
-
+@dataclass
+class WorkflowRun:
+    name: str
+    status: str
+    conclusion: str
+    timestamp: datetime
+    url: str
+
+    @classmethod
+    def parse_from_github(cls, workflow: dict) -> WorkflowRun:
```

Comment:
For some reason this felt better than defining `__init__`. Though now that I think about it, there's not really a case where we'd create one of these NOT from a github api response... Open to thoughts

Reply:
def ➕ to not not futzing with __init__. Could also just move this constructor out of the class (removing the need to import __future__)

Reply:
I personally like static class constructor functions. Just field like `WorkflowRun.parse` reads better than `github.parse_workflow_run`

I'll add the frozen!